### PR TITLE
Add Design Team Operational Guidelines

### PR DESCRIPTION
**Pull request by AlicjaSuska**
I've summarised the rules we've talked about during the last Design Club and created guidelines that we can refer to when making difficult design decisions. Please, share your feedback! I would love the whole Design Team to really feel that those rules are aligned with our values and goals. Thank you!

### CODE REVIEW

File: handbook/product/design/index.md [hunk 1]
```diff
@@ -67,6 +67,34 @@ While design does adds time at the beginning of a project, it improves velocity
 
 Our design process is documented here: **[Design process](design_process.md)**
 
+## Design Team Operational Guidelines
+
+Rules that help us make decisions in crucial moments of the design process.
+
+- **Always design with the advanced user in mind**
+
+Our users are mostly highly skilled developers who are used to complex interfaces. They are more knowledgeable and need less guidance than regular web users. They also have higher expectations and are less forgiving.
```
Comment (christinaforney) [inline handbook/product/design/index.md:76]: Sourcegraph is for all developers, so want to clean up wording to not alienate anyone :) 

```suggestion
Our users are developers who are accustomed to using complex interfaces. They are more knowledgeable and need less guidance than regular web users. They also have higher expectations and are less forgiving.
```

File: handbook/product/design/index.md [hunk 2]
```diff
@@ -67,6 +67,34 @@ While design does adds time at the beginning of a project, it improves velocity
...
+
+- **Consider familiar patterns first**
+
+Whenever possible, use well-known patterns from products our users spend a lot of time in, like IDEs or code hosts.
```
Comment (christinaforney) [inline handbook/product/design/index.md:80]:
```suggestion
Whenever possible, use well-known patterns from products our users spend a lot of time in, like IDEs, code hosts, or other developer tools.
```

File: handbook/product/design/index.md [hunk 3]
```diff
@@ -67,6 +67,34 @@ While design does adds time at the beginning of a project, it improves velocity
...
+
+People are visual creatures. That remains true for our technical audience as well. Sourcegraph interface should feel pleasing, not overwhelming, and natural to use.
+
+- **Design today with the future in mind**
+
+Sourcegraph is growing at an extraordinary pace. Our designs need to be flexible enough to accommodate changes and improvements. Always choose the solutions that work today but can be easily adapted for the future development. Our goal is to redesign as little as possible so our users do not have to adjust to frequent changes.
+
+- **Manage complexity**
+
+We need to handle the unavoidable complexity of our product. Our users should see Sourcegraph as a powerful and intuitive tool. Aim to remove obstacles and unnecessary elements from both flows and interfaces. Use just-in-time knowledge and contextual help to educate and empower. Enable shortcuts and customisations, especially for power users.
```
Comment (christinaforney) [inline handbook/product/design/index.md:96]:
```suggestion
We need to handle the unavoidable complexity of our product. Our users should see Sourcegraph as a powerful and intuitive tool. Aim to remove obstacles and unnecessary elements from both flows and interfaces. Use just-in-time knowledge and contextual help to educate and empower. Enable shortcuts and customizations, especially for technical users accustomed to using keyboard navigation.
```

**Review by christinaforney — APPROVED** [review]

**Comment by AlicjaSuska** [comment]
Thank you for all the suggestions @christinaforney 🎉 @rrhyne @quinnkeast do you have any thoughts on this proposition?

File: handbook/product/design/index.md [hunk 4]
```diff
@@ -67,6 +67,34 @@ While design does adds time at the beginning of a project, it improves velocity
 
 Our design process is documented here: **[Design process](design_process.md)**
 
+## Design Team Operational Guidelines
```
Comment (quinnkeast) [inline handbook/product/design/index.md:70]:
```suggestion
## Design Principles
```

I'd prefer to rename this to "Design principles" – "Operational guidelines" seems more like "the tools, processes we use" than "how we make decisions."

Comment (rrhyne) [reply handbook/product/design/index.md:70]: I agree operational guidelines would describe the process surrounding design at Sourcegraph. This is focused on how we make decisions about design.

**Review by rrhyne — APPROVED** [review]

**Review by quinnkeast — APPROVED** [review]

**Review by rrhyne — APPROVED** [review]

**Pull request merged**